providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -272,7 +272,7 @@ public int compare(Hardware h1, Hardware h2) {
                     .compare(getBootableDeviceType(h1), getBootableDeviceType(h2));
             if (!volumes1.isEmpty() && !volumes2.isEmpty() && volumes1.size() == volumes2.size()) {
```

Comment:
@andreaturli Outside the scope of this commit, but this can't be right -- two otherwise identical objects, one with and without a volume, will compare to zero instead of -1 or +1.